pkg/mediatype/mediatype.go

```diff
@@ -0,0 +1,291 @@
+package mediatype
+
+import (
+	"errors"
+	"mime"
+	"strings"
+
+	mapset "github.com/deckarep/golang-set"
+	"golang.org/x/net/html/charset"
+	"golang.org/x/text/encoding"
+	"golang.org/x/text/encoding/htmlindex"
+)
+
+// MediaType represents a document format, identified by a unique RFC 6838 media type.
+// [MediaType] handles:
+//  - components parsing – eg. type, subtype and parameters,
+//  - media types comparison.
+//
+// Comparing media types is more complicated than it looks, since they can contain parameters,
+// such as `charset=utf-8`. We can't ignore them because some formats use parameters in their
+// media type, for example `application/atom+xml;profile=opds-catalog` for an OPDS 1 catalog.
+//
+// Specification: https://tools.ietf.org/html/rfc6838
+type MediaType struct {
+	Type       string            // The type component, e.g. `application` in `application/epub+zip`.
+	SubType    string            // The subtype component, e.g. `epub+zip` in `application/epub+zip`.
+	Parameters map[string]string // The parameters in the media type, such as `charset=utf-8`.
+
+	// Private until used
+	// name          string
+	fileExtension string
+}
+
+// Create a new MediaType.
+// When an error is returned, do not use the resulting MediaType, as it will be incomplete/invalid
+func NewMediaType(str string, name string, extension string) (mt MediaType, err error) {
```

Comment:
Since we have a dedicated `mediatype` package, we can simplify a bit the name of the constructors (as well as the `MediaTypeOfX` sniffing functions).

```suggestion
func New(str string, name string, extension string) (mt MediaType, err error) {
```

From the coding convention:
> When a function in package `pkg` returns a value of type `pkg.Pkg` (or `*pkg.Pkg`), the function name can often omit the type name without confusion
> https://go.dev/blog/package-names